frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,167 @@
+      try {
+        return await post('/api/v1/namespaces', newNamespaceData, true, {
+          cluster: clusterData || '',
+        });
+      } catch (error) {
+        console.error('Error creating namespace', error);
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        if (errorMessage.includes('already exists')) {
+          setNamespaceDialogOpen(true);
+          setNamespaceNameValid(false);
+          setNameHelperMessage(t('translation|Namespace name already exists.'));
+        }
+        throw error;
+      }
+    }
+
+    setNamespaceDialogOpen(false);
+
+    dispatch(
+      clusterAction(() => namespaceRequest(), {
+        startMessage: t('translation|Applying {{  newItemName  }}…', {
+          newItemName: newNamespaceName,
+        }),
+        cancelledMessage: t('translation|Cancelled applying  {{  newItemName  }}.', {
+          newItemName: newNamespaceName,
+        }),
+        successMessage: t('translation|Applied {{ newItemName }}.', {
+          newItemName: newNamespaceName,
+        }),
+        errorMessage: t('translation|Failed to create {{ kind }} {{ name }}.', {
+          kind: 'namespace',
+          name: newNamespaceName,
+        }),
+        cancelCallback: () => {
+          setNamespaceDialogOpen(true);
+        },
+      })
+    );
+  }
+
+  useEffect(() => {
+    setNamespaceNameValid(Namespace.isValidNamespaceFormat(namespaceName));
+
+    if (!Namespace.isValidNamespaceFormat(namespaceName)) {
+      if (namespaceName.length > 63) {
+        setNameHelperMessage(t('translation|Namespaces must be under 64 characters.'));
+      } else {
+        setNameHelperMessage(
+          t(
+            "translation|Namespaces must contain only lowercase alphanumeric characters or '-', and must start and end with an alphanumeric character."
+          )
+        );
+      }
+    }
+  }, [namespaceName]);
+
+  return (
+    <>
+      <ActionButton
+        color="primary"
+        description={t('translation|Create')}
+        icon={'mdi:plus-circle'}
+        onClick={() => {
+          setNamespaceDialogOpen(true);
+        }}
+      />
+
+      <Dialog
+        aria-label="Dialog"
+        open={namespaceDialogOpen}
+        onClose={() => setNamespaceDialogOpen(false)}
+      >
+        <DialogTitle>{t('translation|Create Namespace')}</DialogTitle>
+        <DialogContent>
+          <Box component="form" style={{ width: '20vw', maxWidth: '20vw' }}>
+            <TextField
+              margin="dense"
+              id="name"
+              aria-label="Name"
+              label={t('translation|Name')}
+              type="text"
+              error={!namespaceNameValid && namespaceName.length > 0}
+              helperText={!namespaceNameValid && namespaceName.length > 0 ? nameHelperMessage : ''}
+              fullWidth
+              value={namespaceName}
+              onChange={event => setNamespaceName(event.target.value)}
```

Comment:
We could lowercase the name of the namespace here. So users would have it automatically done.